desktop/ui/src/main/java/org/datacleaner/windows/DatahubDatastoreDialog.java

```diff
@@ -0,0 +1,351 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.windows;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.ItemEvent;
+import java.awt.event.ItemListener;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import javax.swing.JButton;
+import javax.swing.JCheckBox;
+import javax.swing.JComboBox;
+import javax.swing.JComponent;
+import javax.swing.JOptionPane;
+import javax.swing.JPasswordField;
+import javax.swing.border.EmptyBorder;
+import javax.swing.event.DocumentEvent;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.metamodel.util.FileHelper;
+import org.datacleaner.bootstrap.DCWindowContext;
+import org.datacleaner.bootstrap.WindowContext;
+import org.datacleaner.connection.DatahubDatastore;
+import org.datacleaner.guice.Nullable;
+import org.datacleaner.metamodel.datahub.DatahubConnection;
+import org.datacleaner.user.MutableDatastoreCatalog;
+import org.datacleaner.user.UserPreferences;
+import org.datacleaner.user.UserPreferencesImpl;
+import org.datacleaner.util.DCDocumentListener;
+import org.datacleaner.util.IconUtils;
+import org.datacleaner.util.ImmutableEntry;
+import org.datacleaner.util.LookAndFeelManager;
+import org.datacleaner.util.StringUtils;
+import org.datacleaner.util.WidgetFactory;
+import org.datacleaner.util.WidgetUtils;
+import org.datacleaner.util.http.MonitorHttpClient;
+import org.datacleaner.widgets.DCLabel;
+import org.jdesktop.swingx.JXTextField;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+
+public class DatahubDatastoreDialog
```

Comment:
Renamed all classes to be "DataHubSomething". Updated all references in the project.